src/dftbp/wavegrid/basis/lut.F90

```diff
@@ -0,0 +1,139 @@
+
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2025  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+!> Holds TRadialTableOrbital, a concrete implementation of TOrbital utilising a 1d interpolated lookup table.
+!> This allows for arbitrary radial functions.
+module dftbp_wavegrid_basis_lut
+  use dftbp_common_accuracy, only : dp
+  use dftbp_io_message, only : error
+  use dftbp_wavegrid_basis_orbital, only : TOrbital
+  implicit none
+
+  private
+
+  public :: TRadialTableOrbital, TRadialTableOrbital_initFromArray, TRadialTableOrbital_initFromOrbital
+  
+  !> Concrete class for an orbital represented by a linearly
+  !! interpolated lookup table.
+  type, extends(TOrbital) :: TRadialTableOrbital
+    !> Grid spacing (resolution)
+    real(dp) :: gridDist
+
+    !> Inverse of the grid spacing
+    real(dp) :: invLutStep
+
+    !> Orbital values on the grid
+    real(dp), allocatable :: gridValue(:)
+  contains
+    procedure :: getRadial => TRadialTableOrbital_getRadial
+  end type TRadialTableOrbital
+
+contains
+
+  !> Initialises using a verbatim array of values.
+  subroutine TRadialTableOrbital_initFromArray(this, gridValue, gridDist, angMom)
+    !> TRadialTableOrbital instance to initialise
+    type(TRadialTableOrbital), intent(out) :: this
+
+    !> To be interpolated radial value array
+    real(dp), intent(in) :: gridValue(:)
+
+    !> Grid spacing (resolution)
+    real(dp), intent(in) :: gridDist
+
+    !> Angular momentum of the orbital (l)
+    integer, intent(in) :: angMom
+
+    real(dp) :: cutoff
+
+    this%angMom = angMom
+    this%gridDist = gridDist
+    this%invLutStep = 1.0_dp / gridDist
+
+    cutoff = this%gridDist * real(size(gridValue) - 1, dp)
+    this%cutoffSq = cutoff**2
+
+    allocate(this%gridValue, source=gridValue)
+
+  end subroutine TRadialTableOrbital_initFromArray
+
```

Comment:
Please 2 newlines between subroutines (similar to black-formatting in Python :wink: )